docs/breaking.md

```diff
@@ -0,0 +1,29 @@
+# Breaking changes (migration guide)
+
+This document lists changes introduced in bpftrace which break backwards
+compatibility in some way. Each entry should contain:
+- a link to the PR introducing the change
+- a brief description of the change
+- an example of an error message
+- a simple guide to fix existing scripts
+
+### `pid` and `tid` builtins return `uint32`
+
+https://github.com/bpftrace/bpftrace/pull/3441
+
+Previously, `pid` and `tid` builtins returned `uint64` so it is now possible to
+get an error when storing the builtin in a variable and overriding it with
+`uint64` later:
+```
+# bpftrace -e 'BEGIN { $x = pid; $x = uid; }'   # uid is uint_64
```

Comment:
`uid` might not be a very good example - it really should be a uint32 like pid and we could change that in the future too

Reply:
Good point, changed to `cgroup` which is `uint64` (returned by the BPF helper so very unlikely to change).